Provide more developer value via preferred header

macyabbey opened the pull request:
I've been delighted by `wiretap.` 

As I've used it, I've often wished it could easily mock negative cases that are traditionally hard to test in services which are meant to not fail.

With a small change here to the mock server's algorithm for deciding selection of a mock response, we can unlock even more value from wiretap by mocking negative cases.

daveshanley commented:
> I've been delighted by `wiretap.`
> 
> As I've used it, I've often wished it could easily mock negative cases that are traditionally hard to test in services which are meant to not fail.
> 
> With a small change here to the mock server's algorithm for deciding selection of a mock response, we can unlock even more value from wiretap by mocking negative cases.

I am so glad you're delighted! 

My ultimate goal is this: "Creating software that is beautiful and easy to use. We believe that software should be enjoyable to use, and that it should inspire us."

It's very rewarding to see it manifest. 

I love the addition, thank you. Once the test passes, I can merge it in!

macyabbey commented on mock/mock_engine.go:
```diff
@@ -355,6 +356,10 @@ func (rme *ResponseMockEngine) findMediaTypeContainingNamedExample(
                 responseBody = resp.Content.GetOrZero("application/json")
             }
 
+            if responseBody == nil {
```
handle case that could have panic'd in last rev

macyabbey commented on mock/mock_engine.go:
```diff
@@ -381,14 +386,15 @@ func (rme *ResponseMockEngine) findLowestSuccessCode(operation *v3.Operation) st
 }
 
-func (rme *ResponseMockEngine) lookForResponseCodes(
+func (rme *ResponseMockEngine) findBestMediaTypeMatch(
```
nit: was "looking for" media type objects, and the other functions start with "find" so tweaked the name

macyabbey commented:
> I love the addition, thank you. Once the test passes, I can merge it in!

Yay! Fixed the test and added another test case for a code branch that looked like it could panic to me.
